lib/spack/spack/solver/asp.py

```diff
@@ -2433,14 +2446,23 @@ def visit(node):
         path = os.path.join(parent_dir, "concretize.lp")
         parse_files([path], visit)
 
+    def add_runtime_virtuals(self):
+        """Explicitly add the virtuals that might be attached dynamically by runtimes"""
+        for vpkg in ("fortran-rt", "ifcore", "gfortran"):
+            self.possible_virtuals.add(vpkg)
+            self.gen.fact(fn.possible_in_link_run(vpkg))
+
     def define_runtime_constraints(self):
         """Define the constraints to be imposed on the runtimes"""
         recorder = RuntimePropertyRecorder(self)
         for compiler in self.possible_compilers:
-            if compiler.name != "gcc":
+            if compiler.name not in ("gcc", "oneapi"):
                 continue
+
+            compiler_with_different_cls_names = {"oneapi": "intel-oneapi-compilers"}
```

Comment:
This mapping should live in one place ... ideally not here. Don't we already have it somewhere for bootstrapping compilers the old way?  In any case, this and `possible_compilers` should be the only special cases IMO.

Reply:
We have this in `lib/spack/spack/compilers/__init__.py`.

Reply:
This still needs to be consolidated - it should come from `compilers/__init__.py`